src/cdf.rs

```diff
@@ -5075,66 +5085,40 @@ unsafe fn get_qcat_idx(q: c_int) -> c_int {
     return 3 as c_int;
 }
 
-pub unsafe fn rav1d_cdf_thread_init_static(cdf: *mut CdfThreadContext, qidx: c_int) {
-    (*cdf).r#ref = 0 as *mut Rav1dRef;
-    (*cdf).data.qcat = get_qcat_idx(qidx) as c_uint;
+pub unsafe fn rav1d_cdf_thread_init_static(qidx: c_int) -> CdfThreadContext {
+    CdfThreadContext::QCat(get_qcat_idx(qidx) as c_uint)
 }
 
-pub unsafe fn rav1d_cdf_thread_copy(dst: *mut CdfContext, src: &CdfThreadContext) {
-    if !((*src).r#ref).is_null() {
-        memcpy(
-            dst as *mut c_void,
-            (*src).data.cdf as *const c_void,
-            ::core::mem::size_of::<CdfContext>(),
-        );
-    } else {
-        (*dst).m = av1_default_cdf.clone();
-        (*dst).kfym = default_kf_y_mode_cdf;
-        (*dst).coef = av1_default_coef_cdf[(*src).data.qcat as usize].clone();
-        (*dst).mv.joint = default_mv_joint_cdf;
-        (*dst).dmv.joint = default_mv_joint_cdf;
-        (*dst).dmv.comp[1] = default_mv_component_cdf.clone();
-        (*dst).dmv.comp[0] = (*dst).dmv.comp[1].clone();
-        (*dst).mv.comp[1] = (*dst).dmv.comp[0].clone();
-        (*dst).mv.comp[0] = (*dst).mv.comp[1].clone();
-    };
+pub unsafe fn rav1d_cdf_thread_copy(dst: &mut CdfContext, src: &CdfThreadContext) {
```

Comment:
Could it, probably, but does it need to be? It could also be a method, etc. There's plenty of things we could do here but I'd like to just move on right now.